model/proxy/alphav1/config/cfg.proto

```diff
@@ -21,230 +21,162 @@ package istio.proxy.v1alpha.config;
+// talking to, on behalf of the source service. There can be one or more
+// service versions for a given service (see the discussion on versions above).
+// The proxy would choose the version based on various routing rules.
+//
+// Applications address only the destination service using without knowledge of
+// individual service versions. The actual choice of the version is determined
+// by the proxy, enabling the application code to decouple itself from the
+// evolution of dependent services.
 
 // Proxy level global configurations go here
-message ProxyConfig {
-  // config specification version.
-  int32 revision = 1;
-
-  // Traffic routing (http|l4) related configurations for the
-  // proxy/sidecar.  The configuration generated for the proxy will
-  // preserve the order of rules specified by the user in the Istio config.
-
-  // Requests can match either a l4 or Http (l7) rule, but not both. A
-  // request can match multiple Http (l7) rules, but it can match only one
-  // L4 rule. This can be enforced by checking the match attributes. A Http
-  // match rule matches on both L4 and L7 attributes, while a L4 rule
-  // matches based only on L4 attributes. Any duplication or collision in
-  // the match attributes between two L4 rules or between a L4 and L7 rule
-  // should throw an error.
-
-  // Since a Http request can match multiple route rules (e.g., there are
-  // two routes for the prefix /foo, with differing Http match attributes
-  // such as Http headers), the first rule to match will/should be chosen
-  // by proxy implementations N.B.: When a request does not match any rule
-  // for Http, the downstream service would receive a HTTP 404.
-  repeated RouteRule route_rules = 2;
-  
-  // RouteRule determines the upstream cluster to which traffic should be
-  // routed to.  The UpstreamCluster describes how to obtain cluster
-  // members (through discovery or through static configuration), and
-  // policies that determine how to handle traffic (load balancing
-  // policies, failure recovery policies such as timeouts, retries, circuit
-  // breakers), etc.
-  repeated UpstreamCluster upstream_clusters = 3;
+message ProxyMeshConfig {
 }
 
-// Describes rules for routing a request/connection to an upstream
-// service, based on attributes associated with the request/connection
-// and downstream service invoking the API call. The choice of the specific
-// upstream cluster will be determined by the routing rule.
-message RouteRule {
-  // Uniquely identifies the upstream service (not its versions) associated
-  // with this RouteRule. This is applicable for both Http and Tcp.
-  // Note: The name used here must match with the 'name' field in the
-  // ClusterIdentifier, that is used to uniquely identify a version of the
-  // upstream service.
+// Destination declares policies that determine how to handle traffic for a
+// destination service (load balancing policies, failure recovery policies such
```

Comment:
Kuat, thinking more about this, the problem with the current t destination is that its equivalent to envoy's virtual host. All routes under that host get the same policy. This might not be okay in several cases, especially in cases of retries. The retry policy is very much dependent on the API being called. Payments/checkcredit can be retried for any error code but payments/makepayment can't be retried if destination returns a 500 or example. That's why Envoy has retry policy per route entry. (Timeout as well). In your previous version, iirc destination was equivalent to the upstream cluster with tag based qualifiers and destination name. Right? Why was this removed here?